pretender.js

```diff
@@ -5,18 +5,86 @@ var RouteRecognizer = isNode ? require('route-recognizer')['default'] : window.R
 var FakeXMLHttpRequest = isNode ? require('./bower_components/FakeXMLHttpRequest/fake_xml_http_request') : window.FakeXMLHttpRequest;
 var slice = [].slice;
 
-function Pretender(/* routeMap1, routeMap2, ...*/){
-  maps = slice.call(arguments);
-  // Herein we keep track of RouteRecognizer instances
-  // keyed by HTTP method. Feel free to add more as needed.
-  this.registry = {
+
+/**
+ * parseURL - decompose a URL into its parts
+ * @param  {String} url a URL
+ * @return {Object} parts of the URL, including the following
+ *
+ * 'https://www.yahoo.com:1234/mypage?test=yes#abc'
+ * 
+ * {
+ *   host: 'www.yahoo.com:1234',
+ *   protocol: 'https:',
+ *   search: '?test=yes',
+ *   hash: '#abc',
+ *   href: 'https://www.yahoo.com:1234/mypage?test=yes#abc',
+ *   pathname: '/mypage',
+ *   fullpath: '/mypage?test=yes'
+ * }
+ */
+function parseURL(url) {
+  // TODO: something for when document isn't present... #yolo
+  var anchor = document.createElement('a');
+  anchor.href = url;
+  anchor.fullpath = anchor.pathname + (anchor.search || '') + (anchor.hash || '');
+  return anchor;
+}
+
+
+/**
+ * Registry
+ * 
+ * A registry is a map of HTTP verbs to route recognizers. 
+ */
+
+function Registry(host) {
+  this.verbs = {
     GET: new RouteRecognizer(),
     PUT: new RouteRecognizer(),
     POST: new RouteRecognizer(),
     DELETE: new RouteRecognizer(),
     PATCH: new RouteRecognizer(),
-    HEAD: new RouteRecognizer()
+    HEAD: new RouteRecognizer(),
+    OPTIONS: new RouteRecognizer()
```

Comment:
not sure if anyone is doing anything crazy with options, but imo it doesn't do any harm to add it